core/src/pow/types.rs

```diff
@@ -239,9 +239,19 @@ impl Default for ProofOfWork {
 	}
 }
 
-impl ProofOfWork {
-	/// Read implementation, can't define as trait impl as we need a version
-	pub fn read(_ver: u16, reader: &mut dyn Reader) -> Result<ProofOfWork, ser::Error> {
```

Comment:
I don't think you can do this. You're going to be adding that version back for our scheduled hard fork, aren't you?

Reply:
I don't think the check is going to happen here during deserialization. 

We have `consensus::valid_header_version()` and this is happening orthogonal to the ser/deser process (called via `pipe::validate_header()`).

I _think_ we originally had this passed in here when we first introduced the AR/AT dual PoW impl or when we opened up the C31+ flexibility.

That said - this change was one of the things I wanted to discuss via this PR.

Reply:
I don't know much about the format of our PoW, but I assumed the protocol bump would change how we have to serialize/deserialize ProofOfWork. If that's not true, then I agree with your change.

Reply:
I'd leaning toward simplifying and standardizing how we `read` (to use our existing traits) for now. If and when we discover this needs to be adjusted we can revisit this decision.

Reply:
Ideally the PoW change wouldn't change its format, but just details about graph construction or cycle finding. Not sure we'll be able to keep it that way every time but still a good goal to start with.